contribute/design/1.x/ndm/20200219-ndm-disk-accessing.md

```diff
@@ -0,0 +1,184 @@
+---
+oep-number: NDM 0002
+title: Uniquely accessing disk in cluster
+authors:
+  - "@mittachaitu"
+owners:
+  - "@gila"
+  - "@kmova"
+  - "@vishnuitta"
+  - "@pawanpraka1"
+  - "@akhilerm"
+editor: "@mittachaitu"
+creation-date: 2020-02-19
+last-updated: 2020-02-19
+---
+
+# Disk Identification using NDM
+
+## Table of Contents
+
+- [Disk Identification using NDM](#disk-identification-using-ndm)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Current Implementation](#current-implementation)
+      - [Shortcomings of current implementation](#shortcomings-of-current-implementation)
+      - [Proposed Implementation](#proposed-implementation)
+        - [Workflow](#workflow)
+        - [Advantages](#advantages)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks](#drawbacks)
+  - [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+This proposal brings out the design details to implement a uniquely access the
+uniquely identified disks across the cluster.
+
+## Motivation
+
+The uniquely identifying disk is addressed via PR https://github.com/openebs/openebs/pull/2666
+Now applications required some mechanism to access uniquely identified disk
+across the clusters. This accessing is necessary, due to the fact that disk can
+move from one node to node and even disk can be attached back to different ports of
+the same node. The unique accessing solution will help the consumers of NDM to
+access the disks uniquely even if path got changed.
+
+### Goals
+
+- Configurable device accessing mechanism.
+- A unique disk accessing mechanism that will work across the clusters.
+
+### Non-Goals
+
+- NA
+
+## Proposal
+
+### User Stories
```

Comment:
what about user stories like:
- removing the requirement to run in Privileged mode for NDM consumers
- removing the need for restart of app pod to access disks that are attached after non-privileged pod came up